Gems/Atom/RPI/Code/Tests/Model/ModelTests.cpp

```diff
@@ -1306,4 +1344,71 @@ namespace UnitTest
     }
+
+    // test to verify that each secondary sub meshes are still intersected with correctly when using brute-force
+    // ray intersection
+    class BruteForceMultiModelIntersectsFixture : public ModelTests
+    {
+    public:
+        inline static const float QuadOffsetX = 15.0f;
+
+        void SetUp() override
+        {
+            ModelTests::SetUp();
+            m_mesh = AZStd::make_unique<TestMesh>();
+
+            AZ::RPI::ModelLodAssetCreator lodCreator;
+            m_mesh->Begin(lodCreator);
+
+            // take default quad positions and offset in X by set amount
+            AZStd::vector<float> offsetQuadPositions;
+            offsetQuadPositions.resize(QuadPositions.size());
+            AZStd::copy(QuadPositions.begin(), QuadPositions.end(), offsetQuadPositions.begin());
+            for (size_t xVertIndex = 0; xVertIndex < offsetQuadPositions.size(); xVertIndex += 3)
+            {
+                offsetQuadPositions[xVertIndex] += QuadOffsetX;
+            }
+
+            // create shared buffer to store cube and quad mesh in the same buffer
+            const size_t indicesCount = QuadIndices.size() + CubeIndices.size();
+            const size_t positionCount = QuadPositions.size() + CubePositions.size();
+            AZ::Data::Asset<AZ::RPI::BufferAsset> indexBuffer = BuildTestBuffer(aznumeric_cast<uint32_t>(indicesCount), sizeof(uint32_t));
+            AZ::Data::Asset<AZ::RPI::BufferAsset> positionBuffer =
+                BuildTestBuffer(aznumeric_cast<uint32_t>(positionCount / 3), sizeof(float) * 3);
+
+            // add the cube mesh
+            m_mesh->Add(
+                lodCreator, CubePositions.data(), CubePositions.size(), 0, positionBuffer, CubeIndices.data(), CubeIndices.size(), 0,
+                indexBuffer);
+            // add the quad mesh (offset by the cube position and index data into the same buffer)
+            m_mesh->Add(
+                lodCreator, offsetQuadPositions.data(), offsetQuadPositions.size(), /*offset=*/CubePositions.size() / 3, positionBuffer,
+                QuadIndices.data(), QuadIndices.size(), /*offset=*/CubeIndices.size(), indexBuffer);
+
+            m_mesh->End(lodCreator);
+        }
+
+        void TearDown() override
+        {
+            m_mesh.reset();
+            ModelTests::TearDown();
+        }
+
+        AZStd::unique_ptr<TestMesh> m_mesh;
+    };
+
+    TEST_F(BruteForceMultiModelIntersectsFixture, RayIntersectsWithSecondMesh)
+    {
+        float t = 0.0f;
+        AZ::Vector3 normal = AZ::Vector3::CreateOne(); // invalid starting normal
+
+        // fire a ray at the second sub mesh and ensure a successful hit is returned
+        constexpr bool AllowBruteForce = false;
+        EXPECT_THAT(
+            m_mesh->GetModel()->LocalRayIntersectionAgainstModel(
+                AZ::Vector3(QuadOffsetX, 0.0f, 5.0f), -AZ::Vector3::CreateAxisZ(10.0f), AllowBruteForce, t, normal),
+            testing::Eq(true));
```

Comment:
Nit: You can use `testing::IsTrue()` here:
```suggestion
            testing::IsTrue());
```

Reply:
Sure I can update that :+1: